lib/puppet/network/http/compression.rb

```diff
@@ -20,7 +20,8 @@ module Active
     def uncompress_body(response)
       case response['content-encoding']
       when 'gzip'
-        return Zlib::GzipReader.new(StringIO.new(response.body)).read
+        # ZLib::GzipReader has an associated encoding, by default Encoding.default_external
+        return Zlib::GzipReader.new(StringIO.new(response.body), :encoding => Encoding::BINARY).read
```

Comment:
Totally tangential, but FYI - I just checked RubyMine 2016.3, and out of the box there are no code inspections for not having explicitly specified an encoding.

That said, it looks like you can build custom inspections for IntelliJ based IDEs, so there's some hope we can do something easily - https://www.jetbrains.com/help/idea/2016.3/creating-custom-inspections.html.